src/Http/Controller/SearchController.php

```diff
@@ -7,18 +7,50 @@
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Annotation\Route;
 
+use App\Domain\Course\Entity\Technology;
+use App\Http\Normalizer\TechnologyPathNormalizer;
+
+
 class SearchController extends AbstractController
 {
     /**
      * @Route("/search", name="search")
      */
-    public function search(Request $request, SearchInterface $search): Response
-    {
+    public function search(
+      Request $request,
+      SearchInterface $search,
+      TechnologyPathNormalizer $technologyPathNormalizer
+    ): Response {
         $q = $request->query->get('q');
 
-        return $this->render('pages/search.html.twig', [
-            'q' => $q,
-            'results' => $search->search($q, [])['hits'],
-        ]);
+        // Si q n'est pas vide
+        if ( !empty( $q ) ) {
```

Comment:
Je pense que c'est mieux de rediriger vers une page d'erreur au début, cela est compliqué.
```suggestion
        // On redirige si la recherche est vide
        if (empty($q)) {
            return $this->render('404.html.twig');
        }
```